src/Microsoft.AspNet.Mvc.Core/DefaultControllerTypeProvider.cs

```diff
@@ -76,17 +77,45 @@ protected internal virtual bool IsController([NotNull] TypeInfo typeInfo)
             {
                 return false;
             }
-            if (typeInfo.Name.Equals("Controller", StringComparison.OrdinalIgnoreCase))
+            if (typeInfo.Name.Equals(ControllerTypeName, StringComparison.OrdinalIgnoreCase))
             {
                 return false;
             }
-            if (!typeInfo.Name.EndsWith("Controller", StringComparison.OrdinalIgnoreCase) &&
-                !typeof(Controller).GetTypeInfo().IsAssignableFrom(typeInfo))
+            if (!typeInfo.Name.EndsWith(ControllerTypeName, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
It already is since you also do a name check in it. Maybe break the method into two pieces? One that validates the `IsAssignableFrom` and the other the name.